common/crypto/dkg/src/bte/proof_chunking.rs

```diff
@@ -0,0 +1,1084 @@
+// Copyright 2022 - Nym Technologies SA <contact@nymtech.net>
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::bte::encryption::Ciphertexts;
+use crate::bte::{Chunk, PublicKey, Share, CHUNK_SIZE, NUM_CHUNKS};
+use crate::ensure_len;
+use crate::error::DkgError;
+use crate::utils::{deserialize_g1, hash_to_scalar};
+use crate::utils::{deserialize_scalar, RandomOracleBuilder};
+use bls12_381::{G1Projective, Scalar};
+use ff::Field;
+use group::{Group, GroupEncoding};
+use rand::Rng;
+use rand_core::{RngCore, SeedableRng};
+
+const CHUNKING_ORACLE_DOMAIN: &[u8] =
+    b"NYM_COCONUT_NIDKG_V01_CS01_SHA-256_CHACHA20_CHUNKING_ORACLE";
+
+const SECOND_CHALLENGE_DOMAIN: &[u8] =
+    b"NYM_COCONUT_NIDKG_V01_CS01_WITH_BLS12381_XMD:SHA-256_SSWU_RO_PROOF_CHUNKING_SECOND_CHALLENGE";
+
+/// The number of parallel runs batched in a single challenge,
+/// `l` ($\ell$) in the DKG paper.
+const PARALLEL_RUNS: usize = 32;
+
+/// `lambda` ($\lambda$) in the DKG paper
+const SECURITY_PARAMETER: usize = 256;
+
+// note: ceiling in integer division can be achieved via q = (x + y - 1) / y;
+/// ceil(SECURITY_PARAMETER / PARALLEL_RUNS) in the paper
+const NUM_CHALLENGE_BITS: usize = (SECURITY_PARAMETER + PARALLEL_RUNS - 1) / PARALLEL_RUNS;
+
+// type alias for ease of use
+type FirstChallenge = Vec<Vec<Vec<u64>>>;
+
+// TODO: perhaps break it down into separate arguments after all
+#[cfg_attr(test, derive(Clone))]
+pub struct Instance<'a> {
+    /// y_1, ..., y_n
+    public_keys: &'a [PublicKey],
+
+    /// R_1, ..., R_m
+    randomizers_r: &'a [G1Projective; NUM_CHUNKS],
```

Comment:
This naming of the value is a bit misleading: the actual randomizer value is r, while R is the public value associated with it (you can think about it as a commitment to r or a public value associated with r). 
Note: In general, I would suggest we try in our whole codebase (for when we implement cryptographic primitives and schemes) to have a consistent notation for capital letters and lower-case letters. Sadly, somehow Rust doesn't seem to be happy about using capital letters